R/server.R

```diff
@@ -551,6 +551,28 @@ runApp <- function(appDir=getwd(),
     handlerManager$clear()
   }, add = TRUE)
 
+  appParts <- as.shiny.appobj(appDir)
+
+  if (missing(port)) {
```

Comment:
Are these `missing()` checks necessary? In general, it's best to avoid using `missing()`, because it can make it hard to call the function programmatically.